### DISCUSSION TITLE
How to fetch autoincrement value for a model?

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#7913] [Q&A] [answered]
I have a model that starts like this
![image](https://user-images.githubusercontent.com/40667360/123648245-cb706700-d7f6-11eb-97e5-72e6e375ccbe.png)

How would I fetch the value of the auto-increment?

**Comment by ryands17** [comment, marked as answer]
@Helixable 👋 

I don't think this is directly possible with Prisma but could you explain your use case so that I can provide an alternative?

**Reply by ghost** [reply]
I have a ticket system for a chat program. Users can open tickets and I want to be able to access the ticket number (id) before actually creating the ticket in the database if that makes sense.

**Reply by janpio** [reply]
The requirement is clear, but it does not make much sense. In the time between you get that number, someone else could open a ticket already and the number you are displaying to this user would already be gone. The only way around that would be to block creation of issues after you showed someone else their future ID - which creates a lot of other problems and is probably not what you want in the first place.

If you really need this feature, create an empty `issue` entry to get the ID, then later update it with content or delete it if not issue is acutally opened by the user.

**Reply by Arturo-Lopez** [reply]
The simple solution would be get the last record from the table and then add one.
If your db its going to have a lot of traffic its not approach that i would recommend
because it can generate id collisions at some point. Depending of your db engine
theres some documentation in the subject

Like this case [Determine next auto_increment value before an INSERT in Postgres](https://stackoverflow.com/questions/46172772/determine-next-auto-increment-value-before-an-insert-in-postgres) 

You may need to use the raw sql implementation of prisma for your use case. The engine prisma
does not handle the autoincrement instead it relies on the database for that\

I encourage you see a work around to this problem as I am not a expert is this topic

**Reply by ghost** [reply]
@Arturo-Lopez The problem with your solution is that fact that tickets can be deleted from the database. So the "last" ticket could potentially be deleted and you wouldn't have the right id. @janpio Thank you, I will do what you suggested.

**Reply by tomhoule** [reply]
Another common approach would be using UUIDs or similar random ids that can be generated client-side instead of auto-incrementing integers.
